bin/main.ml

```diff
@@ -1,18 +1,56 @@
 open Diffcessible
 open Cmdliner
+open ExtUnix.Specific
+
+let create_empty_diff_content () =
+  "diff --git a/Error b/Error\n--- a/Error\n+++ b/Error\n"
+
+let setup_input = function
+  | Some path ->
+      if not (Sys.file_exists path) then failwith ("File not found: " ^ path);
+      let ic = In_channel.open_bin path in
+      (Some ic, Notty_unix.Term.create ())
+  | None ->
+      let tty_path = ttyname Unix.stdout in
+      let tty_fd = Unix.openfile tty_path [ Unix.O_RDWR ] 0o600 in
+      let term = Notty_unix.Term.create ~input:tty_fd ~output:tty_fd () in
+      if not (Unix.isatty Unix.stdin) then (Some In_channel.stdin, term)
+      else (None, term)
+
+let safe_read_input ic =
+  try
+    let content = In_channel.input_all ic in
+    In_channel.close ic;
+    content
+  with _ ->
+    In_channel.close ic;
+    create_empty_diff_content ()
 
 let main file_path =
-  let file = In_channel.open_bin file_path in
-  let s = In_channel.input_all file in
-  let patch = Patch.to_diffs s in
-  Interactive_viewer.start patch
+  let ic_opt, term = setup_input file_path in
+  let input_content =
+    match ic_opt with
+    | None -> create_empty_diff_content ()
+    | Some ic -> safe_read_input ic
```

Comment:
That might close stdin depending on what `setup_input` returns.

Reply:
Sorry can I clarify what you mean here? I thought `safe_read_input` handles closing stdin when it is opened?

Reply:
`safe_read_input` always closes the channel but depending on which branch is taken in `setup_input`, the channel might be `stdin`, which shouldn't get closed.